src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -41,6 +42,79 @@ def persist_flow_run(self, run_info: FlowRunInfo):
         self.queue.put(run_info)
 
 
+class HealthyEnsuredProcess:
+    def __init__(self, executor_creation_func):
+        self.process = None
+        self.input_queue = None
+        self.output_queue = None
+        self.is_ready = False
+        self._executor_creation_func = executor_creation_func
+
+    def start_new(self):
+        input_queue = Queue()
+        output_queue = Queue()
+        self.input_queue = input_queue
+        self.output_queue = output_queue
+
+        # Put a start message and wait the subprocess be ready.
+        # Test if the subprocess can receive the message.
+        input_queue.put("start")
+
+        current_log_context = LogContext.get_current()
+        process = Process(
+            target=_process_wrapper,
+            args=(
+                self._executor_creation_func,
+                input_queue,
+                output_queue,
+                current_log_context.get_initializer() if current_log_context else None,
+                OperationContext.get_instance().get_context_dict(),
+            ),
+            # Set the process as a daemon process to automatically terminated and release system resources
+            # when the main process exits.
+            daemon=True
+        )
+
+        self.process = process
+        process.start()
+
+        try:
+            # Wait for subprocess send a ready message.
+            ready_msg = output_queue.get(timeout=30)
+            logger.info(f"Process {process.pid} get ready_msg: {ready_msg}")
+            self.is_ready = True
+        except queue.Empty:
+            logger.info(f"Process {process.pid} did not send ready message, exit.")
+            self.end()
+            self.start_new()
+
+    def end(self):
+        # When process failed to start and the task_queue is empty.
+        # The process will no longer re-created, and the process is None.
+        if self.process is None:
+            return
+        if self.process.is_alive():
+            self.process.kill()
+
+    def put(self, args):
+        self.input_queue.put(args)
+
+    def get(self):
+        return self.output_queue.get(timeout=1)
+
+    def format_current_process(self, line_number: int, is_completed=False):
```

Comment:
I didn't find a case where `is_complete` is True.

Reply:
fixed